### PR TITLE
Remove case sensitivity in hostname check

### PR DESCRIPTION
**Pull request by huguesBouvier**
Series of fix + improvement on iotedge check.
PR following: https://github.com/Azure/iotedge/pull/4608

Removing case sensitivity when checking hostname.
It created issue, like being unable to use fqdn if your hostname has a capital letter (if you use azure VMs).

### CODE REVIEW

**Comment by massand** [comment]
The hostname is needed to generate server certificates with the right DNS entries needed for mutual TLS auth by leaf devices. So, I believe this requirement cannot be changed in 1.2 GA. 

The capitalization issues with hostname in checks can be resolved with `to_lowercase`

**Comment by massand** [comment]
Please change the PR description to match the change - i.e. fixed case sensitivity. Thanks.

File: edgelet/iotedge/src/check/checks/hostname.rs [hunk 1]
```diff
@@ -86,8 +86,12 @@ impl Hostname {
         //
         // Instead, we punt on this check and assume that everything's fine if config_hostname is identical to the device hostname,
         // or starts with it.
+        //
+        // Azure FQDN don't support capital letter so we lower case the hostname before doing the check.
         if config_hostname != machine_hostname
```
Comment (massand) [inline edgelet/iotedge/src/check/checks/hostname.rs:91]: to_lowercase() these as well.

Comment (huguesBouvier) [reply edgelet/iotedge/src/check/checks/hostname.rs:91]: Ok, though that does't change the logic.

**Review by massand — APPROVED** [review]

**Pull request merged**